timesheets/Code.gs

```diff
@@ -0,0 +1,149 @@
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+ 
+// Global variables representing the index of certain columns.
+var COLUMN_NUMBER = {
+  EMAIL: 3,
+  HOURS_START: 4,
+  HOURS_END: 8,
+  HOURLY_PAY: 9,
+  CALC_PAY: 10,
+  APPROVAL: 11,
+};
+
+/**
+* Creates the menu item "Timesheets" so user can run scripts on drop-down.
+*/
+function onOpen() {
+    var ui = SpreadsheetApp.getUi();
+    ui.createMenu('Timesheets')
+        .addItem('Column Setup', 'columnSetup')
+        .addItem('Calculate Pay', 'calculatePay')
+        .addItem('Send Emails', 'sendEmails')
+        .addToUi();
+}
+
+/**
+* Runs the "Column Setup" script, adding an "APPROVAL" column at the end of the sheet, 
+* containing drop-down menus to either approve/disapprove employee timesheets.
+* Adds "NOTIFIED STATUS" column indicating whether or not an employee has yet been e mailed.
+*/
+function columnSetup() { 
+  // defines variables
+  var sheet = SpreadsheetApp.getActiveSheet();
+  var lastCol = sheet.getLastColumn();
+  var lastRow = sheet.getLastRow();
+  var frozenRows = sheet.getFrozenRows();
+
+  // adds new calculate pay column
+  sheet.insertColumnAfter(COLUMN_NUMBER.HOURLY_PAY);
+  var weeklyPayColumn = sheet.getRange(1, COLUMN_NUMBER.CALC_PAY).setValue('WEEKLY PAY');
+  
+  // adds new approval column
+  sheet.insertColumnAfter(COLUMN_NUMBER.CALC_PAY);
+  var approvalColumn = sheet.getRange(1, COLUMN_NUMBER.APPROVAL).setValue('APPROVAL');
+
+  // make sure approval column is all drop-down menus
+  var bigRange = sheet.getRange(frozenRows + 1, COLUMN_NUMBER.APPROVAL, lastRow - frozenRows, 1);  
+  var dropdownValues = ['APPROVED', 'NOT APPROVED', 'IN PROGRESS'];
+  var rule = SpreadsheetApp.newDataValidation().requireValueInList(dropdownValues).build();
+  bigRange.setDataValidation(rule);
+  bigRange.setValue('IN PROGRESS');
+  
+  // adds new notified column
+  sheet.insertColumnAfter(COLUMN_NUMBER.APPROVAL); //global
+  var notifiedColumn = sheet.getRange(1, COLUMN_NUMBER.APPROVAL + 1).setValue('NOTIFIED STATUS'); 
+  
+  // make sure notified column is all drop-down menus
+  bigRange = sheet.getRange(frozenRows + 1, COLUMN_NUMBER.APPROVAL + 1, lastRow - frozenRows, 1);
+  dropdownValues = ['NOTIFIED', 'NOT NOTIFIED'];
+  rule = SpreadsheetApp.newDataValidation().requireValueInList(dropdownValues).build();
+  bigRange.setDataValidation(rule);
+  bigRange.setValue('NOT NOTIFIED');
+}
+
+/**
+* Runs the "Calculate Pay" script, creating a new column with each employee's 
+* pay based on their reported hours and hourly wage.
+*/
+function calculatePay() {
```

Comment:
I don't think it's realistic to use a script to calculate the pay when it's much easier to do it with a formula. You could just place this formula in the first top cell in the column:
```
=ArrayFormula(SUM(C2:G)*H2:H)
```
You could do that once in the setup function.

Reply:
Seem to have overlooked this - what do you think? Just wrote CodeLab for this solution talking about the calculatePay traversing through rows etc. which could be useful to show? Otherwise will change, let me know!

Reply:
Editing now - on it !